src/updaters/krm/krm-blueprint-version.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Update, UpdateOptions, VersionsMap} from '../update';
+import {GitHubFileContents} from '../../github';
+import {logger} from '../../util/logger';
+
+export class KRMBlueprintVersion implements Update {
+  path: string;
+  changelogEntry: string;
+  version: string;
+  versions?: VersionsMap;
+  packageName: string;
+  create: boolean;
+  contents?: GitHubFileContents;
+
+  constructor(options: UpdateOptions) {
+    this.create = false;
+    this.path = options.path;
+    this.changelogEntry = options.changelogEntry;
+    this.version = options.version;
+    this.packageName = options.packageName;
+  }
+  updateContent(content: string): string {
+    // js-yaml(and kpt TS SDK) does not preserve comments hence regex match
+    // match starting cnrm/ ending with semver to prevent wrong updates like pinned config.kubernetes.io/function
+    const oldVersion = content.match(
```

Comment:
We might need to be a bit more careful here. In particular, if a package contains a subpackage pulled from another repository (at an older version), we *don't* want to rewrite that version number. Instead I think we might need to just search for the exact previous version and replace with that.

Reply:
It looks like the released package is also on the line to replace, so you might be able to match on that as well.

Reply:
@morgante @chingor13 I added some logic to use previous tag if available. We still fallback to default matching if no tag available (IIUC, there would be a bootstrapping problem otherwise on first release).

> released package is also on the line to replace

I will investigate this in a follow up. We need to align some of our existing attributions with pkg names in the upstream repo.

Reply:
Thanks, this new approach looks workable.